### PR TITLE
Feature/at 8046 task order details clin table

### PR DESCRIPTION
**Pull request by DGabriel-CCPO**
Users are to view details on their task order.

### CODE REVIEW

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue [hunk 1]
```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
...
+  public async createTableData(): Promise<void>{
+    await this.collectTableData();
+    await this.sortRows();
+    await this.addSeparators();
+    /// create table data
+    this.tableData = [...this.tableData,...this.optionPendingClins, ...this.expiredClins]
+  }
+
+
+  public calculateTotalFundingObj(clin: ClinTableRowData): void{
```
Comment (ajaynalamala-ccpo) [inline src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:537]: This function name threw me off a little. Initially I thought all the calculation is happening inside this function. But up on deep dive, I saw that this function does iterative amendment to the total. A better name?

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue [hunk 2]
```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
...
+    if (clin.isActive || clin.isExercised) {
+      const totalCLINValue = currencyStringToNumber(clin.totalCLINValue || "0");
+      const obligatedFunds = currencyStringToNumber(clin.obligatedFunds || "0");
+      const totalFundsSpent = currencyStringToNumber(clin.totalFundsSpent || "0");
```
Comment (ajaynalamala-ccpo) [inline src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:541]: For a task order, the obligate funds and totalFundsSpent should be available from the store. Do we need to calculate them here?

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue [hunk 3]
```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
...
+    await this.sortRows();
+    await this.addSeparators();
+    /// create table data
+    this.tableData = [...this.tableData,...this.optionPendingClins, ...this.expiredClins]
+  }
+
+
+  public calculateTotalFundingObj(clin: ClinTableRowData): void{
+    if (clin.isActive || clin.isExercised) {
+      const totalCLINValue = currencyStringToNumber(clin.totalCLINValue || "0");
```
Comment (ajaynalamala-ccpo) [inline src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:539]: A "TODO" to remove this line after feedback from the BA? Or a TODO to remove this function completely based on whether you need the below two lines?

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue [hunk 4]
```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
...
+    }
+    let percent = Math.round((Number(fundsSpent) / Number(obligatedFunds)) * 100);
+    let remaining = Number(obligatedFunds) - Number(fundsSpent);
+    return {
+      percent: String(percent),
+      fundsRemaining:"$" + toCurrencyString(remaining) + " remaining"
+    }
+  }
+
+  public timeToExpiration(start:string,end:string): Record<string, string> {
```
Comment (ajaynalamala-ccpo) [inline src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:440]: Can we move this function and the respective template code into it own util component? Component that takes in "start" and "end" date properties and does the rest and displays. This change will reduce the lines in this already large component by about 60 lines and potentially any duplications of this elsewhere.

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue [hunk 5]
```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
...
+      if (this.selectedTaskOrder.totalObligated && this.selectedTaskOrder.totalFundsSpent) {
+        this.taskOrderRemainingFunds = this.fundsRemaining(
+          currencyStringToNumber(this.selectedTaskOrder.totalObligated), 
+          currencyStringToNumber(this.selectedTaskOrder.totalFundsSpent),
+        );
+      }
+
+    } 
+    catch {
+      console.log("Error loading Task Order Details")
```
[the remark this answers is not in the document]
Comment (ericyoungquist-ccpo) [reply src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:596]: it is intentional

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue (hunk 4 above)
Comment (ericyoungquist-ccpo) [reply src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:440]: good idea. moved to `helpers/index.ts`

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue (hunk 1 above)
Comment (ericyoungquist-ccpo) [reply src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:537]: @ajaynalamala-ccpo The name is misleading I agree. It previously calculated BOTH active totals and the total Task Order values, but then after discussing with you that the T.O. values were already calculated in the store, I refactored to use those. Will rename to be less confusing.

File: src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue (hunk 2 above)
Comment (ericyoungquist-ccpo) [reply src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue:541]: see above reply - this is only calculating current period totals

**Review by ajaynalamala-ccpo — APPROVED** [review]

**Pull request merged**